### PR TITLE
Ingredient

### PR DESCRIPTION
**Pull request by IanSylvester**

### CODE REVIEW

File: api/models/Category.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require("mongoose");
+
+const categorySchema = new mongoose.Schema(
```
Comment (Doode1524) [inline api/models/Category.js:3]: schema should be capitalized: ie CategorySchema

File: api/models/Category.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    {
+    name: {
+        type: String,
+        required: true,
+        minlength: 3,
+        maxlength: 255
+    },
+});
+
+module.exports = mongoose.model("Category", categorySchema);
```
Comment (Doode1524) [inline api/models/Category.js:13]: Same here, capitalize to match

File: api/models/Ingredient.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const mongoose = require("mongoose");
+
+const ingredientSchema = new mongoose.Schema(
```
Comment (Doode1524) [inline api/models/Ingredient.js:3]: shema should be capitalized

File: api/models/Ingredient.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            ref: "Category"
+        },
+        subCategory: {
+            type: Schema.Types.ObjectId,
+            ref: "SubCategory"
+        }
+          
+    });
+
+mongoose.model("Ingredient", ingredientSchema);
```
Comment (Doode1524) [inline api/models/Ingredient.js:42]: same here to match

File: api/models/SubCategory.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const mongoose = require("mongoose");
+
+const SubCategorySchema = new mongoose.Schema({
```
Comment (Doode1524) [inline api/models/SubCategory.js:3]: What is this Model going to be used for?

File: api/models/User.js [hunk 6]
```diff
@@ -6,12 +6,38 @@ const UserSchema = new mongoose.Schema(
...
+    Organisation: {
+      type: new mongoose.Schema({
+        name: {
+          type: String,
+          required: true,
+          minLength: 3,
+          maxLength: 255
+        }
+      })
+    }
```
Comment (Doode1524) [inline api/models/User.js:40]: Organization is spelled wrong and should be it's own model. We will add a relationship with the user later

File: api/models/Vendor.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+
+const mongoose = require("mongoose");
+
+const vendorSchema = new mongoose.Schema({
```
Comment (Doode1524) [inline api/models/Vendor.js:4]: Capital for schema here too

File: api/models/Vendor.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        minlength: 3,
+        maxlength: 255
+    },
+    vendorRep: {
+        type: Schema.Types.ObjectId,
+        ref: "VendorRep"
+    },
+});
+
+module.exports = mongoose.model("Vendor", vendorSchema);
```
Comment (Doode1524) [inline api/models/Vendor.js:33]: and here to match

File: api/models/Vendor.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        minlength: 3,
+        maxlength: 255
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true
+    },
+    phone: {
+        type: String,
```
Comment (Doode1524) [inline api/models/Vendor.js:17]: phone number would be type number, or whatever mongoose has it. Maybe integer

Comment (Doode1524) [reply api/models/Vendor.js:17]: unless you want to format it, then it can be a string. Ie: "352-476-8348"

File: api/models/VendorRep.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+        maxlength: 255 
+    },
+    email: {
+        type: String, 
+        required: true, 
+        unique: true
+    },
+    phone: {
+        type: String, 
+        required: true,
```
Comment (Doode1524) [inline api/models/VendorRep.js:17]: same as above. Up to you

File: api/models/Ingredient.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+            ref: "Vendor"
+        },
+        category: {
+            type: Schema.Types.ObjectId,
+            ref: "Category"
+        },
+        subCategory: {
+            type: Schema.Types.ObjectId,
+            ref: "SubCategory"
+        }
```
Comment (Doode1524) [inline api/models/Ingredient.js:38]: are the refs for referencing other models?

File: api/models/Ingredient.js [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+            required: true
+        },   
+        productBrand: {
+            type: String,
+            required: false
+        },
+        productSKU: {
+            type: String,
+            required: false,
+        },
```
Comment (Doode1524) [inline api/models/Ingredient.js:26]: Do SKUs contain characters or letters? If not, this should probably be a number type

File: api/models/SubCategory.js (hunk 5 above)
Comment (IanSylvester) [reply api/models/SubCategory.js:3]: sub category of a product. I.E. Category is Liquor, sub categories will be Whiskey, Vodka, Rum etc

File: api/models/Vendor.js (hunk 9 above)
Comment (IanSylvester) [reply api/models/Vendor.js:17]: I was planning on formating

File: api/models/Ingredient.js (hunk 11 above)
Comment (IanSylvester) [reply api/models/Ingredient.js:38]: yes, was hoping that I did this right...

File: api/models/Ingredient.js (hunk 12 above)
Comment (IanSylvester) [reply api/models/Ingredient.js:26]: some SKUs have both

File: api/models/SubCategory.js (hunk 5 above)
Comment (Doode1524) [reply api/models/SubCategory.js:3]: Let's leave it for now, but we may have to rethink later. SubCategory will probably end up being a joins table

File: api/models/Ingredient.js (hunk 11 above)
Comment (Doode1524) [reply api/models/Ingredient.js:38]: Vendor should definitely be it's own model. Same as above, I think we can leave it for now, and figure out our relationships later. We will need joins tables

**Pull request merged**